File: 9.贪心算法/17.用最少数量的箭引爆气球/452.用最少数量的箭引爆气球.py
#
# @lc app=leetcode.cn id=452 lang=python3
#
# [452] 用最少数量的箭引爆气球
#
from mytools import * 
# @lc code=start
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[0])
        if not points: return 0
        res = 1
        n = len(points)
        for i in range(1, n):
            if points[i][0] > points[i - 1][1]:
                res += 1
            else:
                points[i][1] = min(points[i][1], points[i - 1][1])
        return res
# @lc code=end

File: 9.贪心算法/17.用最少数量的箭引爆气球/452.用最少数量的箭引爆气球.cpp
/*
 * @lc app=leetcode.cn id=452 lang=cpp
 *
 * [452] 用最少数量的箭引爆气球
 */
#include "tools.h"
// @lc code=start
class Solution {
public:
    static bool cmp(vector<int>& a, vector<int>& b)
    {
        return a[0] < b[0];
    }
    int findMinArrowShots(vector<vector<int>>& points) {
        ranges::sort(points, cmp);
        int res = 1;
        int n = points.size();
        for (int i = 1; i < n; i ++ )
        {
            if (points[i][0] > points[i - 1][1]) res ++;
            else points[i][1] = min(points[i][1], points[i - 1][1]);
        }
        return res;
    }
};
// @lc code=end

